### DISCUSSION TITLE
How does DeployRestic flag works if volume snapshot location is also configured

### DISCUSSION THREAD

**Discussion by pradeepkchaturvedi** [vmware-tanzu/velero#4118] [Community support Q&A] [answered]
If I installed Velero with below options,
- DeployRestic = True 
- DefaultSnapshotLocation is configured 
-  snapshotsEnabled = true

In this case, would it be Provider snapshot or Restic backup for PVs. I could not find details about this case in docs. PV provider matches the Snapshot location config, would Velero first try provider snapshot, and if that fails then fall back to Restic or it would only use Restic in such case.

**Comment by reasonerjt** [comment, marked as answer]
If the `--default-volumes-to-restic` is not set. 
You need to explicitly annotate the pod to be backed up by Restic:
https://velero.io/docs/v1.6/restic/#using-opt-in-pod-volume-backup

Once a pod is set to be backed up by Restic, the provider snapshot will be skipped:
https://github.com/vmware-tanzu/velero/blob/dbc83af77b0c0a36a27ac1646b6504fa78bedaba/pkg/backup/item_backupper.go#L422-L425

**Reply by pradeepkchaturvedi** [reply]
@reasonerjt Thanks for the answer. Please verify below

1. If cluster has multiple volumes and  --default-volumes-to-restic is not set. 
- If user annotates only few pod volumes then those having annotation will get backedup using Restic and for other pod volumes provider snapshot will be tried.
-  If none of the pod volumes has annotations, then for all volumes provider snapshot will be tried.

2. 1. If cluster has multiple volumes and  --default-volumes-to-restic is set
- In this case user need not annotate any pod volume all volumes will backed up using Restic. Use may Opt-out for pod volumes not to be backed up using Restic as mentioned at https://velero.io/docs/v1.6/restic/#using-the-opt-out-approach
- In this scenario, pod volumes with opt-out annotation, would Velero try provider snapshot for them? or those volumes will be skipped from backup.

**Reply by reasonerjt** [reply]
I have verified the scenarios you mentioned they worked as you expected.

**Comment by reasonerjt** [comment]
> If cluster has multiple volumes and --default-volumes-to-restic is set
> 
> In this scenario, pod volumes with opt-out annotation, would Velero try provider snapshot for them? or those volumes will be skipped from backup.

Velero will try provider snapshot.

**Reply by pradeepkchaturvedi** [reply]
@reasonerjt  Thank you for verifying it.
